pilot/pkg/xds/filters/filters.go

```diff
@@ -138,3 +163,50 @@ var (
 	mtlsHTTP11ALPN = []string{"istio-http/1.1", "istio"}
 	mtlsHTTP2ALPN  = []string{"istio-h2", "istio"}
 )
+
+func buildHTTPMxFilter() *hcm.HttpFilter {
+	vmConfig := &v3.PluginConfig_VmConfig{
+		VmConfig: &v3.VmConfig{
+			Runtime: "envoy.wasm.runtime.null",
+			Code: &core.AsyncDataSource{Specifier: &core.AsyncDataSource_Local{
+				Local: &core.DataSource{
+					Specifier: &core.DataSource_InlineString{
+						InlineString: "envoy.wasm.metadata_exchange",
+					},
+				},
+			}},
+		},
+	}
+
+	if features.EnableWasm {
+		vmConfig = &v3.PluginConfig_VmConfig{
+			VmConfig: &v3.VmConfig{
+				Runtime:          "envoy.wasm.runtime.v8",
+				AllowPrecompiled: true,
+				Code: &core.AsyncDataSource{Specifier: &core.AsyncDataSource_Local{
+					Local: &core.DataSource{
+						Specifier: &core.DataSource_Filename{
+							Filename: "/etc/istio/extensions/metadata-exchange-filter.compiled.wasm",
+						},
+					},
+				}},
+			}}
+	}
+
+	httpMxConfigProto := &wasm.Wasm{
+		Config: &v3.PluginConfig{
+			Vm:            vmConfig,
+			Configuration: util.MessageToAny(&protobuf.StringValue{Value: "{}\n"}),
```

Comment:
```suggestion
			Configuration: util.MessageToAny(&protobuf.StringValue{Value: "{}"}),
```

nit but no need to include a newline